test/specs/routing/body-matching.test.js

```diff
@@ -112,6 +113,54 @@ describe('body matching', () => {
 		expect(fm.calls(true).length).to.equal(1);
 	});
 
+	it('should match exact FormData bodies', async () => {
+		const formData = new FormData();
+		formData.append('foo', 'bar');
+
+		fm.mock({ body: formData }, 200).catch();
+
+		await fm.fetchHandler('http://a.com/', {
+			method: 'POST',
+			body: formData,
+			headers: {},
+		});
+		expect(fm.calls(true).length).to.equal(1);
+	});
+
+	it('should not match FormData bodies that are not the same', async () => {
+		const expectedFormData = new FormData();
+		expectedFormData.append('foo', 'bar');
+
+		fm.mock({ body: expectedFormData }, 200).catch();
+
+		const actualFormData = new FormData();
+		actualFormData.append('fizz', 'buzz');
+		await fm.fetchHandler('http://a.com/', {
+			method: 'POST',
+			body: actualFormData,
+			headers: {},
+		});
+		expect(fm.calls(true).length).to.equal(0);
+	});
+
+	it('should not ignore the order of the form fields', async () => {
```

Comment:
We might be able to do something more complicated here, but I was a touch worried about the semantics of having multiple values for the same keys/arrays/etc.

We'd probably want to convert the `FormData` objects to JS objects and compare those as we are today (with `isEqual`) instead?